### ISSUE TITLE
Fish shell support

### ISSUE THREAD

**Issue by xixixao** [trogdoro/xiki#108]
Any plans to support fish? Anyone else has any experience integrating the two? Eh?

**Comment by trogdoro** [comment]
Can you try fish shell with the latest version of xiki on github?

**Comment by smith** [comment]
I don't think it works as of 39a692eb408757d488f1b2837ed285c7bc0eb59e. Doing the initial setup for me added `source ~/.xsh` to my ~/.bashrc. Sourcing that (or xiki-master/bin/.xsh) won't work because they are sh scripts that fish doesn't understand.

Probably wouldn't be too hard to either port them to fish (which would not be great if they needed to change at some point) or figure out how to have fish handle them in their current form (which I don't think is possible, but there could be ways to do it.)

**Comment by atourkow** [comment]
+1

**Comment by seanmadsen** [comment]
+1

**Comment by laughedelic** [comment]
@smith probably [edc/bass](https://github.com/edc/bass) could be useful here?

**Comment by saintromuald** [comment]
Has anyone been able to make this work now? I haven't found anything regarding that.

**Comment by yozlet** [comment]
I have it _partially_ working, using the aforementioned `bass` in a new function:

```fish
function xsh
	bass source $HOME/.xsh ';' xsh --d $argv
end
```

Note that this doesn't set up key shortcuts for fish, but commands like `xsh +note` work fine. The `--d` starts emacs & xiki in ["isolated"](https://github.com/trogdoro/xiki/blob/master/bin/xsh#L386) mode, without daemons running.
